### PR TITLE
Create dummy updatedatabase worker

### PR DESCRIPTION
**Pull request by anvial**
The PR adds a dummy upgradedatabase worker. This worker unlocks
the database update complete lock. This is the first step to upgradedatabse
worker reviving.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
go test github.com/juju/juju/worker/upgradedatabase/... -gocheck.v

juju bootstrap lxd lxd
juju switch controller
juju ssh 0

source /etc/profile.d/juju-introspection.sh
juju_engine_report | grep -A 10 upgrade-database-
```

## Links

**Jira card:** [JUJU-[4655]](https://warthogs.atlassian.net/browse/JUJU-4655)

### CODE REVIEW

File: worker/upgradedatabase/package_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+package upgradedatabase
+
+import (
+	"testing"
+
+	gc "gopkg.in/check.v1"
+)
+
+//go:generate go run go.uber.org/mock/mockgen -package upgradedatabase -destination worker_test.go github.com/juju/juju/worker/upgradedatabase Logger
```
Comment (manadart) [inline worker/upgradedatabase/package_test.go:12]: Generate this into package_mock_test.go, and the other one into lock_mock_test.go.

We'll be using worker_test.go for actual tests.

File: worker/upgradedatabase/manifold.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+// Logger defines the logging methods used by the worker.
+type Logger interface {
+	Criticalf(string, ...interface{})
+	Errorf(string, ...interface{})
+	Warningf(string, ...interface{})
+	Infof(string, ...interface{})
+	Debugf(string, ...interface{})
+	Tracef(string, ...interface{})
+}
```
Comment (SimonRichardson) [inline worker/upgradedatabase/manifold.go:22]: Do we need all of these? 

Also can you change `interface{}` to `any`.

File: worker/upgradedatabase/upgradedatabase.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	if c.Logger == nil {
+		return errors.NotValidf("nil Logger")
+	}
+	return nil
+}
+
+// NewUpgradeDatabaseWorker returns a new Worker.
+func NewUpgradeDatabaseWorker(config Config) (*Worker, error) {
+	var err error
+
```
Comment (SimonRichardson) [inline worker/upgradedatabase/upgradedatabase.go:60]: Delete me.

File: worker/upgradedatabase/manifold.go (hunk 2 above)
Comment (anvial) [reply worker/upgradedatabase/manifold.go]: I've removed Critical and Trace (all others look like used in prev worker mongoDB-version)

File: worker/upgradedatabase/upgradedatabase.go (hunk 3 above)
Comment (anvial) [reply worker/upgradedatabase/upgradedatabase.go]: done

**Pull request merged**